torch/onnx/_internal/fx/passes/readability.py

```diff
@@ -0,0 +1,127 @@
+    are registered at root level.
+    """
+
+    def __init__(
+        self,
+        diagnostic_context: diagnostics.DiagnosticContext,
+        module: torch.fx.GraphModule,
+        original_module: torch.nn.Module,
+    ):
+        super().__init__(diagnostic_context, module)
+        self.original_module = original_module
+
+    @_beartype.beartype
+    def _rename_param_and_buffer(
+        self,
+        diagnostic: diagnostics.Diagnostic,
+        nodes: Sequence[torch.fx.Node],
+        new_name: str,
+    ) -> None:
+        """Rename the parameter/buffer and replace corresponding nodes with new nodes of updated target."""
+        assert len(nodes) > 0, "`nodes` cannot be empty"
+        assert (
+            len({node.target for node in nodes}) == 1
+        ), "`nodes` must all have same `target`"
+        old_name = nodes[0].target
+        assert isinstance(old_name, str), f"Expected str, got type({old_name})"
+        # Parameter/buffer name cannot contain "."
+        # TODO(bowbao): fix #104670 and replace "." with "/" to avoid collision.
+        normalized_name = new_name.replace(".", "_")
+        attr_value = getattr(self.module, old_name)
+        setattr(self.module, normalized_name, attr_value)
+        delattr(self.module, old_name)
+        for node in nodes:
+            with self.module.graph.inserting_before(node):
+                new_node = self.module.graph.get_attr(normalized_name)
+                new_node.meta = node.meta
+                node.replace_all_uses_with(new_node)
+                self.module.graph.erase_node(node)
+        diagnostic.with_additional_message(
+            f"Renamed 'self.{old_name}' to 'self.{normalized_name}', "
+            f"normalized from original parameter name '{new_name}'."
+        )
+
+    def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
+        """Restore parameter and buffer names from original module.
+
+        For each `get_attr` node, if the target is a str representing a parameter or buffer
+        under `self.module`, we rename the parameter or buffer to its original name.
+        The parameters and buffers between `self.module` and `self.original_module` refer
+        to the same objects, allowing us to use it as key to retrieve the original name.
+        """
+        assert len(args) == 0, "RestoreParameterAndBufferNames does not take any args"
+        assert (
+            len(kwargs) == 0
+        ), "RestoreParameterAndBufferNames does not take any kwargs"
+        # state_to_readable_name[parameter/buffer] returns the original readable name of
+        # the parameter/buffer. E.g., "self.linear.weight".
+        state_to_readable_name: Dict[Union[torch.nn.Parameter, torch.Tensor], str] = {}
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_parameters()}
+        )
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_buffers()}
+        )
```

Comment:
Not sure about this one.
* This is not doing name mangling, but rather constructing a reverse lookup table.
* In previous code the replacement happens near the source of that requirement. Putting it here feels arbitrary. 
* If let's say we do need to support params that already exist with format of `xxx.yyy`, we cannot use this lookup table because "." is already replaced.
* Now `.replace(".", "_")` appears twice, then we need another API / lambda just for that. Feels over designing.

Reply:
My point was that `state_to_readable_name` does not store the *final* readable name as it is. It is just a temporary state that will be post processed by yet another function. if we need to know the updated name in another place, we will have to repeat the same search/replace at `state_to_readable_name` to get the final name again

not a blocking though